openedx_external_enrollments/tests/external_enrollments/tests_pathstream_external_enrollments.py

```diff
@@ -29,7 +33,332 @@ def test_str(self):
+        """
+        access_key_id = 'access_key'
+        secret_access_key = 'secret_access_key'
+
+        self.base._init_s3()  # pylint: disable=protected-access
+
+        boto3_mock.client.assert_called_with(
+            's3',
+            aws_access_key_id=access_key_id,
+            aws_secret_access_key=secret_access_key,
+        )
+
+    def test_download_file(self):
+        """
+        _download_file must return the downloaded content from the S3 file if called with all parameters and the
+        s3 connection is established.
+        """
+        s3_file = 'test.log'
+        s3_bucket = 'test'
+        expected_file_content = 'file_content_test'
+        self.base.client = Mock(spec=boto3.client('s3'))
+        self.base.client.get_object.return_value = {
+            'Body': io.StringIO(expected_file_content),
+            'ResponseMetadata': {
+                'HTTPStatusCode': 200,
+            },
+        }
+        log = 'File content [{}] was downloaded from S3 for [{}]'.format(s3_file, self.base.__str__())
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            file_content = self.base._download_file()  # pylint: disable=protected-access
+
+            log_capture.check(
+                (self.module, 'INFO', log),
+            )
+
+        self.base.client.get_object.assert_called_with(
+            Bucket=s3_bucket,
+            Key=s3_file,
+        )
+        self.assertEqual(file_content, expected_file_content)
+
+    def test_download_file_error(self):
+        """If no connection, credentials or wrong parameter values
+        _download_file should handle ClientError and raise PathstreamTaskExecutionError."""
+        code = '400'
+        message = 'Failed'
+        operation = 'get_object'
+        attrs = {
+            'get_object.side_effect': ClientError(
+                {
+                    'Error': {
+                        'Code': code,
+                        'Message': message,
+                    },
+                },
+                operation,
+            ),
+        }
+        clienterror_msg = 'An error occurred ({}) when calling the {} operation: {}'.format(
+            code,
+            operation,
+            message,
+        )
+        self.base.client = Mock(spec=boto3.client('s3'), **attrs)
+
+        with self.assertRaisesMessage(PathstreamTaskExecutionError, clienterror_msg):
+            self.base._download_file()  # pylint: disable=protected-access
+
+    def test_prepare_new_content(self):
+        """This test validates that _prepare_new_content accomplish
+        its task of sorting and encoding."""
+        content = [
+            'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n',
+            'course_T2,2@mail,2021-06-29 16:51:00.456900,true\n',
+            'course_T1,1@mail,2021-06-29 16:50:00.456900,true\n',
+            'course_T2,2@mail,2021-06-29 16:52:00.456900,false\n',
+        ]
+        expected_result = (
```

Comment:
Is this a string? a tuple? What kind of entity do you expect? @Jacatove

Reply:
No, it is a bytes object. actually this `b'course_T1,1@mail,2021-06-29 16:50:00.456900,true\ncourse_T2,2@mail,2021-06-29 16:51:00.456900,true\ncourse_T2,2@mail,2021-06-29 16:52:00.456900,false\ncourse_T1,1@mail,2021-06-29 16:53:00.456900,false\n'` @Squirrel18